### PR TITLE
update traffic policy docs

### PR DESCRIPTION
**Pull request by Megalonia**
update traffic policy docs to include new actions, keywords and user agent filter example

### CODE REVIEW

**Comment by rkolavo** [comment]
Can we somehow reuse the header variable interpolation stuff written for the module in this add-headers page (and also custom-response)?

File: docs/http/traffic-policy/gallery.mdx [hunk 1]
```diff
@@ -72,3 +72,31 @@ This rule applies rate limiting of `30` requests per second to the endpoint `/ap
 		],
 	}}
 />
+
+## User agent filtering
+
+Here we are checking whether the `User-Agent` header has case insensitive `(?i)` string `edg/` followed by some digits `\d`. You can test this by viewing this [regex101 link](https://regex101.com/r/3NPVub/1).
```
Comment (nijikokun) [inline docs/http/traffic-policy/gallery.mdx:78]:
```suggestion
Here we are serving custom content to Microsoft Edge users by checking whether the `User-Agent` header has case insensitive `(?i)` string `edg/` followed by some digits `\d`. You can test this by viewing this [regex101 link](https://regex101.com/r/3NPVub/1).
```

File: docs/http/traffic-policy/actions/custom-response.mdx [hunk 2]
```diff
@@ -18,6 +20,9 @@ configured response.
 
 If this action is executed, no further actions in the traffic policy will be executed.
 
+<VariableInterpolation />
+<HeaderTemplate type="custom-response" />
```
Comment (nijikokun) [inline docs/http/traffic-policy/actions/custom-response.mdx:24]: 🥇 nice

**Comment by Megalonia** [comment]
> Can we somehow reuse the header variable interpolation stuff written for the module in this add-headers page (and also custom-response)?

done!

File: src/components/VariableInterpolation.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+				<code>{templateSyntax}</code> syntax.
+			</p>
+			<p>
+				For example, to include geographical data about the client IP that
+				initiated the request, you may construct a header value like so.
+			</p>
+			<ConfigExample config={{ actions: [{ type, config }] }} />
+			<p>
+				Consult the{" "}
+				<a href="/docs/http/request-headers/#variables">Variables Reference</a>{" "}
```
Comment (Megalonia) [inline src/components/VariableInterpolation.tsx:28]: when I click the link in the preview it flashes "page not found" then redirects to variable refs

Comment (rkolavo) [reply src/components/VariableInterpolation.tsx:28]: oh I was not thinking a link that takes you to the module, I was the variable interpolation documentation on the module page would be broken out into a component that was shared

Comment (Megalonia) [reply src/components/VariableInterpolation.tsx:28]: you want me to take all the variables and add them to this component?

Comment (rkolavo) [reply src/components/VariableInterpolation.tsx:28]: yiss

**Pull request closed without merging**